magefile.go

```diff
@@ -72,3 +85,58 @@ func Notice() error {
 		filepath.Join("dev-tools", "templates", "notice", "NOTICE.txt.tmpl"),
 	)
 }
+
+// GoImports executes goimports against all .go files in and below the CWD.
+func GoImports() error {
+	goFiles, err := FindFilesRecursive(func(path string, _ os.FileInfo) bool {
+		return filepath.Ext(path) == ".go"
+	})
+	if err != nil {
+		return err
+	}
+	if len(goFiles) == 0 {
+		return nil
+	}
+
+	fmt.Println(">> fmt - goimports: Formatting Go code")
+	if err := gotool.Install(
+		gotool.Install.Package(filepath.Join(GoImportsImportPath)),
+	); err != nil {
+		return err
+	}
+
+	args := append(
+		[]string{"-local", GoImportsLocalPrefix, "-l", "-w"},
+		goFiles...,
+	)
+
+	return sh.RunV("goimports", args...)
+}
+
+// FindFilesRecursive recursively traverses from the CWD and invokes the given
+// match function on each regular file to determine if the given path should be
+// returned as a match. It ignores files in .git directories.
+func FindFilesRecursive(match func(path string, info os.FileInfo) bool) ([]string, error) {
+	var matches []string
+	err := filepath.Walk(".", func(path string, info os.FileInfo, err error) error {
```

Comment:
It's a nitpick but anyway can be a big deal on a project like Beats:

> Walk is less efficient than WalkDir, introduced in Go 1.16, which avoids calling os.Lstat on every visited file or directory.

https://pkg.go.dev/path/filepath@go1.19#Walk

Also, it might make sense to read the list of changed files in the last commit and `goimport` them instead of processing all the Go files in the project, so maybe we should not even walk the whole tree.

Reply:
Thanks! I had copied it as is from Beats, but I'll implement this improvement :D